extensions/VisualEditor/wikia/ApiAddMediaTemporary.php

```diff
@@ -154,6 +156,14 @@ protected function checkPermissions() {
 		}
 	}
 
+	protected function checkPermissionsForJA() {
```

Comment:
@owend how do you suggest about checking wgdisableanonymousediting for Japan specific behavior?

For all local except Japan, we do not allow anonymous file upload no matter what. For Japan, we allow it unless wgdisableanonymousediting = true.